core/src/data/value/binary_op/decimal.rs

```diff
@@ -0,0 +1,257 @@
+            I64(other) => *self == Dec::from(*other),
+            F64(other) => *self == Dec::from_f64_retain(*other).unwrap(),
+            Decimal(other) => *self == *other,
+            _ => false,
+        }
+    }
+}
+
+impl PartialOrd<Value> for Dec {
+    fn partial_cmp(&self, other: &Value) -> Option<Ordering> {
+        match *other {
+            I8(rhs) => self.partial_cmp(&(Dec::from(rhs))),
+            I64(rhs) => self.partial_cmp(&(Dec::from(rhs))),
+            F64(rhs) => self.partial_cmp(&(Dec::from_f64_retain(rhs).unwrap())),
+            Decimal(rhs) => self.partial_cmp(&rhs),
+            _ => None,
+        }
+    }
+}
+
+impl TryBinaryOperator for Dec {
+    type Rhs = Value;
+
+    fn try_add(&self, rhs: &Self::Rhs) -> Result<Value> {
+        let lhs = *self;
+
+        match *rhs {
+            I8(rhs) => Ok(Decimal(lhs + Dec::from(rhs))),
+            I64(rhs) => Ok(Decimal(lhs + Dec::from(rhs))),
+            F64(rhs) => Ok(Decimal(lhs + Dec::from_f64_retain(rhs).unwrap())),
+            Decimal(rhs) => Ok(Decimal(lhs + rhs)),
+            Null => Ok(Null),
+            _ => Err(ValueError::AddOnNonNumeric(Decimal(lhs), rhs.clone()).into()),
+        }
+    }
+
+    fn try_subtract(&self, rhs: &Self::Rhs) -> Result<Value> {
+        let lhs = *self;
+
+        match *rhs {
+            I8(rhs) => Ok(Decimal(lhs - Dec::from(rhs))),
+            I64(rhs) => Ok(Decimal(lhs - Dec::from(rhs))),
+            F64(rhs) => Ok(Decimal(lhs - Dec::from_f64_retain(rhs).unwrap())),
+            Decimal(rhs) => Ok(Decimal(lhs - rhs)),
+            Null => Ok(Null),
+            _ => Err(ValueError::SubtractOnNonNumeric(Decimal(lhs), rhs.clone()).into()),
+        }
+    }
+
+    fn try_multiply(&self, rhs: &Self::Rhs) -> Result<Value> {
+        let lhs = *self;
+
+        match *rhs {
+            I8(rhs) => Ok(Decimal(lhs * Dec::from(rhs))),
+            I64(rhs) => Ok(Decimal(lhs * Dec::from(rhs))),
+            F64(rhs) => Ok(Decimal(lhs * Dec::from_f64_retain(rhs).unwrap())),
+            Decimal(rhs) => Ok(Decimal(lhs * rhs)),
+            //Interval(rhs) => Ok(Interval(lhs * rhs)),
+            Null => Ok(Null),
+            _ => Err(ValueError::MultiplyOnNonNumeric(Decimal(lhs), rhs.clone()).into()),
+        }
+    }
+
+    fn try_divide(&self, rhs: &Self::Rhs) -> Result<Value> {
+        let lhs = *self;
+
+        match *rhs {
+            I8(rhs) => Ok(Decimal(lhs / Dec::from(rhs))),
+            I64(rhs) => Ok(Decimal(lhs / Dec::from(rhs))),
+            F64(rhs) => Ok(Decimal(lhs / Dec::from_f64_retain(rhs).unwrap())),
+            Decimal(rhs) => Ok(Decimal(lhs / rhs)),
+            Null => Ok(Null),
+            _ => Err(ValueError::DivideOnNonNumeric(Decimal(lhs), rhs.clone()).into()),
+        }
+    }
+
+    fn try_modulo(&self, rhs: &Self::Rhs) -> Result<Value> {
+        let lhs = *self;
+
+        match *rhs {
+            I8(rhs) => Ok(Decimal(lhs.checked_rem(Dec::from(rhs)).unwrap())),
```

Comment:
This feature looks good too. thank you
Can you return an error instead of `unwrap` when an overflow occurs?
- ex) Add `OverflowError { operator: char }` to the Error enum

Reply:
I've added an Overflow error.   I believe this describes the checked_rem failures.
what do you think would be a good error for when from_f64_retain fails?  F64ToDecimalConversionError?

Reply:
As explained in the following link, 
<https://docs.rs/rust_decimal/latest/rust_decimal/prelude/struct.Decimal.html#method.checked_rem>
it seems that an overflow occurred when trying Decimal % Decimal.

What about messages like `ModuloOverflowError` rather than Conversion?

We don't actually do any conversions in the `check_ram` method.

Reply:
I'm referring to the from_f64_retain function, not checked_rem.    the docs doesn't say anything about overflow or conversion errors for from_f64_retain.

Reply:
Hmm...? You obviously use `checked_rem` in your code snippet.
Conversion related errors are correct by using `from_f64_retain` elsewhere.
But isn't `checked_rem` the one you're currently using?
<img width="799" alt="image" src="https://user-images.githubusercontent.com/27040963/164980817-055e6525-70a7-4622-bca6-8b0e9544ff1f.png">

Reply:
> what do you think would be a good error for when from_f64_retain fails? F64ToDecimalConversionError?

Ah, sorry. I just checked the code snippet and answered it.

In the case of `from_f64_retain` , it would be nice to have a Conversion error message.

Reply:
F64toDecimalConversionError has been added.